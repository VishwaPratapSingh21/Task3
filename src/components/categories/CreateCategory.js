import React, { Component } from "react";
import { notify } from "react-notify-toast";
import axiosInstance from "../Apicalls";

/**
 * Component for allowing users to add categories.
 * @author [Hadijah kyampeire](https://github.com/hadijahkyampeire/Yummy_Reactjs_frontend)
 */
class CreateCategory extends Component {
  state = {
    title: ""
  };

  handleInputChange = event => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  };
  /**
   * handles addition of categories
   * @param {object} event - Reference to the submit form component
   * used to prevent default form behaviours on submission
   */
  handleAddCategories = (event, id) => {
    const { title: name } = this.state;
    event.preventDefault();
    axiosInstance
      .post("categories/", { name })
      .then(response => {
        notify.show(response.data.message, 'success', 4000);
        document.getElementById("closeAddModal").click();

        this.props.getCategories();
        this.setState({ title: "" });
      })
      .catch(error => {
        if (error.response) {
          this.setState({ error: error.response.data.message, mess: "" });
        } else if (error.request) {
          notify.show("Request not made");
        }
      });
  };

  render() {
    return (
      <div>
        <button
          type="button"
          className="btn btn-primary"
          data-toggle="modal"
          data-target="#createCategoryModal"
          aria-hidden="true"
        >
          <i className="fa fa-plus" /> Add category
        </button>
        <div className="modal fade" id="createCategoryModal">
          <div className="modal-dialog" role="document">
            <div className="modal-content">
              <div className="modal-header">
                <h5 className="modal-title">Add category</h5>
                <button
                  type="button"
                  className="close"
                  data-dismiss="modal"
                  aria-label="Close"
                >
                  <span aria-hidden="true">&times;</span>
                </button>
              </div>
              <form onSubmit={this.handleAddCategories}>
                <div className="modal-body">
                  {this.state.error ? (
                    <div className="alert alert-danger">{this.state.error}
                     <button
                    type="button"
                    class="close"
                    data-dismiss="alert"
                    aria-label="Close"
                  >
                    <span aria-hidden="true">&times;</span>
                  </button></div>
                  ) : (
                    ""
                  )}

                  <div className="input-group ">
                    <span className="input-group-addon pr-4">
                      <i className="glyphicon glyphicon-pencil" />
                    </span>
                    <input
                      type="text"
                      name="title"
                      className="form-control"
                      placeholder="category name"
                      onChange={this.handleInputChange}
                      value={this.state.title}
                    />
                  </div>
                </div>
                <div className="modal-footer">
                  <input
                    type="submit"
                    className="btn btn-primary"
                    value="Add"
                  />
                  <button
                    type="button"
                    className="btn btn-secondary"
                    data-dismiss="modal"
                    id="closeAddModal"
                  >
                    Close
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
export default CreateCategory;
