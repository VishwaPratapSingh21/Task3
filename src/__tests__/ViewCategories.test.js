import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson, { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';

import Categories ,{Category} from '../components/categories/Dashboard';
import CreateCategory from '../components/categories/CreateCategory';

describe('ViewCategories component', () => {
  const wrapper = shallow(<Categories />);

  it('renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('renders other components', () => {
    expect(wrapper.find(CreateCategory)).toHaveLength(1);
    expect(wrapper.find('Pagination')).toHaveLength(1);
    expect(wrapper.find('Search')).toHaveLength(1);
  });

  it('has a div', () => {
    expect(wrapper.find('div')).toHaveLength(6);
  });

  it('renders get categories functions', () => {
    expect(wrapper.find('getCategories')).toBeTruthy();
  });

  it('renders delete categpries functions', () => {
    expect(wrapper.find('deleteCategory')).toBeTruthy();
  });

  it('renders delete categories functions', () => {
    expect(wrapper.find('editCategory')).toBeTruthy();
  });

  it('renders change page functions', () => {
    expect(wrapper.find('changePage')).toBeTruthy();
  });

  it('has initial states', () => {
    expect(wrapper.state().categories).toEqual([]);
    expect(wrapper.state().Next_page).toEqual(null);
    expect(wrapper.state().Previous_page).toEqual(null);
    expect(wrapper.state().current_page).toEqual(null);
    expect(wrapper.state().total_pages).toEqual(null);
    expect(wrapper.state().total_Items).toEqual(null);
    expect(wrapper.state().searching).toEqual(false);
  });
});

describe('Category component', ()=>{
  const category ={id:1 , name: "lunch"}
  const wrapper = shallow(<Category {...category} editCategory={jest.fn()} deleteCategory={jest.fn()} />)

  it('renders correctly', ()=>{
    expect(shallowToJson(wrapper)).toMatchSnapshot();
    wrapper.find('.btn-primary .card-link').simulate('click')
    wrapper.find('.btn-danger .card-link').simulate('click')
  })
})