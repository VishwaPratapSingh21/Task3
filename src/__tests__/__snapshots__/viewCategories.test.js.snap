// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category component renders correctly 1`] = `
<div
  className="col-md-4 col-lg-3 col-sm-6 category-card"
>
  <div
    className="card "
  >
    <div
      className="card-block color word-font"
    >
      <h3
        className="card-title"
      >
        lunch
      </h3>
      <div
        className="card-block"
      >
        Date_created: 
        <br />
        Date_modified: 
      </div>
      <div
        className="card-block text-center"
      >
        <Link
          className="btn btn-sm btn-primary card-link"
          data-target="#edit_category1"
          data-toggle="modal"
          replace={false}
          to="#"
        >
          <i
            className="fa fa-edit"
          />
           Edit
        </Link>
        <Link
          className="btn btn-sm btn-danger card-link"
          data-target="#delete_category1"
          data-toggle="modal"
          replace={false}
          to="#"
        >
          <i
            className="fa fa-trash"
          />
           Delete
        </Link>
        <Link
          className="btn btn-sm btn-success card-link"
          replace={false}
          to={
            Object {
              "pathname": "/category/1/recipes",
              "state": Object {
                "categoryName": "lunch",
              },
            }
          }
        >
          <i
            className="fa fa-eye"
          />
           Recipes
        </Link>
      </div>
      <EditCategory
        editCategory={[Function]}
        id={1}
        name="lunch"
      />
      <DeleteCategory
        deleteCategory={[Function]}
        id={1}
        name="lunch"
      />
    </div>
  </div>
</div>
`;

exports[`ViewCategories component renders properly 1`] = `
<div>
  <div
    className="viewcategories"
  >
    <div
      className="row justify-content-md-center"
    >
      <CreateCategory
        getCategories={[Function]}
      />
      <Search
        handleSearch={[Function]}
        placeholder="categories "
      />
    </div>
    <br />
    <div
      className="row categories"
    >
      <div
        className="col-sm-5 offset-sm-3"
      >
        <div
          className="alert alert-info"
          role="alert"
        >
          <strong>
            Ooops!
          </strong>
           No categories , go a head and add some
        </div>
      </div>
    </div>
    <Pagination
      activePage={null}
      changePage={[Function]}
      nextPage={null}
      previousPage={null}
      totalPages={null}
    />
  </div>
</div>
`;
